routers/api/v1/activitypub/reqsignature.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package activitypub
+
+import (
+	"crypto"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"code.gitea.io/gitea/modules/activitypub"
+	gitea_context "code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/modules/httplib"
+	"code.gitea.io/gitea/modules/setting"
+
+	ap "github.com/go-ap/activitypub"
+	"github.com/go-fed/httpsig"
+)
+
+func getPublicKeyFromResponse(b []byte, keyID *url.URL) (p crypto.PublicKey, err error) {
+	person := ap.PersonNew(ap.IRI(keyID.String()))
+	err = person.UnmarshalJSON(b)
+	if err != nil {
+		err = fmt.Errorf("ActivityStreams type cannot be converted to one known to have publicKey property: %v", err)
+		return
+	}
+	pubKey := person.PublicKey
+	if pubKey.ID.String() != keyID.String() {
+		err = fmt.Errorf("cannot find publicKey with id: %s in %s", keyID, string(b))
```

Comment:
I don't think we should be this specific in the error message.
There could be sensitive data in this JSON string we are simply ignoring because it doesn't fit in our data model...

Reply:
If there is sensitive data in that JSON string, that means that the other ActivityPub server is seriously failing their security and making sensitive data publicly accessible in their API. Not our problem here.